src/Aspire.Hosting/Dcp/DcpDependencyCheck.cs

```diff
@@ -0,0 +1,239 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Serialization;
+using Aspire.Hosting.Dcp.Process;
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Globalization;
+using Aspire.Hosting.Properties;
+
+namespace Aspire.Hosting.Dcp;
+
+internal sealed partial class DcpDependencyCheck
+{
+    // Docker goes to into resource saver mode after 5 minutes of not running a container (by default).
+    // While in this mode, the commands we use for the docker runtime checks can take quite some time
+    private const int WaitTimeForDcpInfoCommandInSeconds = 10;
+
+    [GeneratedRegex("[^\\d\\.].*$")]
+    private static partial Regex VersionRegex();
+
+    public static async Task EnsureDcpDependenciesAsync(string? dcpPath, bool hasContainers, string? containerRuntime, CancellationToken cancellationToken)
+    {
+        if (!File.Exists(dcpPath))
+        {
+            throw new FileNotFoundException($"The Aspire application host is not installed at \"{dcpPath}\". The application cannot be run without it.", dcpPath);
```

Comment:
This language may be confusing. AppHost is actually the developers executable. DCP is not the app host.

Reply:
"The Aspire orchestration component is not installed..." maybe?